### PR TITLE
Bugfix miDebuggerPath problem

### PR DESCRIPTION
**Pull request by KoeMai**
## This change addresses item #375 and #242

This refactoring extracts the logic to generate
debugconfigurations from the debug command into a new module.
This new module generates debugConfigurations out of
the cmakecache and support now the generation of miDebuggerPath.
This module is reintegrated into the debug command.

### CODE REVIEW

File: src/debugger.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import {CMakeCache} from '@cmt/cache';
+
+
+export namespace Debugger {
```
Comment (vector-of-bool) [inline src/debugger.ts]: Namespaces are a strange beast. I used them in `vscode-gitflow`, but I didn't find them very useful and just made things cluttered. I'd prefer to just use regular imports instead.

**Comment by KoeMai** [comment]
I removed the not required namespace.
Additional I restore the apple gcc support on mac, but I am not able to try it out (I have no mac).

Do we need some extension tests or integration tests?

**Comment by KoeMai** [comment]
I have no idea how I can make travis happy again. :/

**Comment by vector-of-bool** [comment]
Merge if you are ready.

**Comment by KoeMai** [comment]
I made this modification on windows, there it works. For linux I need to add a fallback scenarios. 
- Clang and gcc-7 does not work, lldb is there a separate package (clang->lldb->gdb)
- gcc-7  has know not it own gdb7 it use the gdb (gcc-7->gcc-7->gdb). 

The integration test does not start the real debugger, thats way travis and appveyor are happy.

**Comment by KoeMai** [comment]
should we add a Setting to set Thema fallback debugger? or should we add a debugger to the kits?

**Pull request closed without merging**